cmd/scheduler/main.go

```diff
@@ -101,6 +106,12 @@ func injectProfilingRoute(router *httprouter.Router) {
 	})
 }
 
+func isLeader() bool {
+	lease, _ := client.GetClient().CoordinationV1().Leases(config.LeaderElectResourceNamespace).
+		Get(context.Background(), config.LeaderElectResourceName, metav1.GetOptions{})
+	return lease.Spec.HolderIdentity != nil && *lease.Spec.HolderIdentity == os.Getenv("HOSTNAME")
+}
+
 func start() error {
```

Comment:
@luohua13 As mentioned in the meeting, i think we can use the `RunOrDie` func from [leaderelection](https://pkg.go.dev/k8s.io/client-go/tools/leaderelection#RunOrDie) package, you can take [this](https://github.com/kubernetes/client-go/blob/master/examples/leader-election/main.go) as example.

Reply:
@luohua13 I had intended to use the **Coordinated Leader Election feature** in [LeaderElectionConfig](https://pkg.go.dev/k8s.io/client-go/tools/leaderelection#LeaderElectionConfig), however i didn't notice that it is still in ALPHA stage. So i suggest that we can add a TODO comment here for future improvement.